threshsign/bench/BenchmarkThreshsign.cpp

```diff
@@ -0,0 +1,279 @@
+// Concord
+//
+// Copyright (c) 2018-2022 VMware, Inc. All Rights Reserved.
+//
+// This product is licensed to you under the Apache 2.0 license (the "License").
+// You may not use this product except in compliance with the Apache 2.0 License.
+//
+// This product may include a number of subcomponents with separate copyright
+// notices and license terms. Your use of these subcomponents is subject to the
+// terms and conditions of the subcomponent's license, as noted in the
+// LICENSE file.
+//
+
+#define PICOBENCH_IMPLEMENT
+#define PICOBENCH_STD_FUNCTION_BENCHMARKS
+
+#include <vector>
+#include <deque>
+#include <cstdlib>
+#include <iostream>
+#include <chrono>
+#include <boost/program_options.hpp>
+#include "thread_pool.hpp"
+
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wconversion"
+#pragma GCC diagnostic ignored "-Wsign-conversion"
+#pragma GCC diagnostic ignored "-Wsign-compare"
+#pragma GCC diagnostic ignored "-Wunused-variable"
+#pragma GCC diagnostic ignored "-Wc99-extensions"
+#include "../../thirdparty/include/picobench.h"
+#include "macros.h"
+#include "include/threshsign/eddsa/EdDSAMultisigFactory.h"
+#include "include/threshsign/eddsa/SingleEdDSASignature.h"
+#include "include/threshsign/bls/relic/BlsThresholdFactory.h"
+#include "include/threshsign/bls/relic/PublicParametersFactory.h"
+
+static std::unique_ptr<std::vector<uint8_t>> s_data{nullptr};
+
+/**
+ * Initializes randomDataBytes random bytes
+ * @param randomDataBytes
+ */
+void initRandomData(size_t randomDataBytes) {
```

Comment:
There is an existing function that generates randomData. `generateRandomData()`.
Can we use it?

Reply:
If we had a library of test utilities then yes. 
Since this is benchmark code and not production code, I didn't think that it matters.